pkg/cache/application_info.go

```diff
@@ -170,3 +171,10 @@ func (ai *ApplicationInfo) GetTag(tag string) string {
 	}
 	return tagVal
 }
+
+func (ai *ApplicationInfo) String() string {
+	ai.lock.Lock()
+	defer ai.lock.Unlock()
```

Comment:
Hi @kingamarton 
I guess this only needs a Read lock, correct?

Reply:
Yes, @yangwwei you are right. I will change it.